tensorflow/python/data/benchmarks/benchmark_base.py

```diff
@@ -124,3 +147,32 @@ def run_and_report_benchmark(self,
     extras["num_elements"] = num_elements
     self.report_benchmark(
         wall_time=wall_time, iters=iters, name=name, extras=extras)
+    return wall_time
+
+  def _consume_dataset(self, dataset, num_elements):
+    """Consume the `num_elements` of the dataset.
+
+    Args:
+      dataset: A tf.data.Dataset to consume data from.
+      num_elements: Number of dataset elements to iterate through.
+    """
+
+    # NOTE: We use `dataset.skip()` to perform the iterations in C++, avoiding
+    # the overhead of having to execute a TensorFlow op for each step of the input
+    # pipeline. Note that this relies on the underlying implementation of `skip`
+    # to execute upstream computation. If it is optimized in the future,
+    # we will have to change this code.
+    dataset = dataset.skip(num_elements)
```

Comment:
`num_elements - 1` to match what the description of the method claims to be doing

Reply:
addressed in the commit